data/styles/clear/include/Roads.mapcss

```diff
@@ -1025,28 +1032,45 @@ line|z19-[highway=steps]
 
 line|z17-[highway=footway][tunnel?]::tunnelBackground,
 line|z17-[highway=path][tunnel?]::tunnelBackground,
-line|z17-[highway=steps][tunnel?]::tunnelBackground,
 {
-  casing-width: eval(prop("width")+0.5);
+  casing-width-add: 0.5;
   casing-linecap: butt;
   casing-color: @bridge_background;
 }
-line|z17-[highway=steps][tunnel?]::tunnelBackground,
-{casing-width: eval(prop("width")/2 + 0.5);}
 
 line|z17-[highway=footway][tunnel?]::tunnelCasing,
 line|z17-[highway=path][tunnel?]::tunnelCasing,
-line|z17-[highway=steps][tunnel?]::tunnelCasing,
 {
-  casing-width: eval(prop("width")+1);
+  casing-width-add: 1;
   casing-linecap: butt;
   casing-color: @bridge_casing;
   casing-dashes: 2,2;
 }
-line|z17-[highway=steps][tunnel?]::tunnelCasing,
-{casing-width: eval(prop("width")/2 + 1);}
 
 
+line|z17-[highway=steps][tunnel?]::tunnelBackground,
+{
+  casing-width: 3.5;
+  casing-linecap: butt;
+  casing-color: @bridge_background;
+}
+line|z18[highway=steps][tunnel?]::tunnelBackground,
+{casing-width: 4.5;}
+line|z19[highway=steps][tunnel?]::tunnelBackground,
+{casing-width: 5.5;}
+
+line|z17-[highway=steps][tunnel?]::tunnelCasing,
+{
+  casing-width: 4;
+  casing-linecap: butt;
+  casing-color: @bridge_casing;
+  casing-dashes: 2,2;
+}
+line|z18[highway=steps][tunnel?]::tunnelCasing,
+{casing-width: 5;}
+line|z19[highway=steps][tunnel?]::tunnelCasing,
+{casing-width: 6;}
```

Comment:
So, why do we manually define casing-width for tunnel here?
It also should depend from highway=steps widht.

Reply:
The original style was `casing-width: eval(prop("width")/2 + 1);`.
I.e. it was based not on `width`, but a half of it (`width/2`).
Hence its not possible to just replace it with `width + casing-width-add` formula and get currently existing final result for all zoom levels.
`width/2` was a kind of exception used in two rules only, so I just replaced them with per-zoom parameters to keep the compiled style unchanged.

Reply:
:) Well, so this is a point where eval is better

Reply:
Yeap it shows its flexibility here.
But I don't think it makes sense to maintain it just for 2 exceptional rules. Moreover its not clear whether this exception was justified at all. Maybe a tunnel with an ordinary `casing-width-add: 1` would look as good or better even - need to test it carefully. My goal here was to make as little real visual changes as possible hence I just replaced the eval with equivalent `casing-width` params.